db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/TranslateFunctionNode.java

```diff
@@ -83,6 +75,34 @@ public ValueNode bindExpression(FromList fromList,
                                     List<AggregateNode> aggregateVector) throws StandardException {
         bindOperands(fromList, subqueryList, aggregateVector);
 
+        for (int i = 0; i < 4; ++i) {
+            if (operands.get(i) != null && operands.get(i).requiresTypeFromContext()) {
+                castOperandAndBindCast(i, DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.VARCHAR));
+            }
+        }
+
+        if (operands.get(0).getTypeId().isBitTypeId()) {
+            throw StandardException.newException(SQLState.NOT_IMPLEMENTED, "TRANSLATE on BIT DATA types not yet implemented");
+        }
+
+        if (!operands.get(0).getTypeId().isCharOrVarChar()) {
+            castOperandAndBindCast(0, DataTypeDescriptor.getBuiltInDataTypeDescriptor(
+                    Types.VARCHAR,
+                    operands.get(0).getTypeServices().isNullable(),
+                    operands.get(0).getTypeCompiler().getCastToCharWidth(operands.get(0).getTypeServices(), getCompilerContext())));
+        }
+
+        // Only one operand: TRANSLATE is equivalent to UPPER
+        if (operands.get(1) == null) {
+            ValueNode upper = new SimpleStringOperatorNode(operands.get(0), "upper", getContextManager());
+            return upper.bindExpression(fromList, subqueryList, aggregateVector);
+        }
+
+        // output but no input table: currently not supported
+        if (operands.get(2) == null) {
+            throw StandardException.newException(SQLState.NOT_IMPLEMENTED, "TRANSLATE(op, outputTranslationTable) not implemented");
+        }
+
         for (int i = 0; i < 3; ++i) {
```

Comment:
This loop seems redundant (and incomplete) now. Shall it be removed?

Reply:
Thanks for the catch! 
Done